packages/common/src/discussions/HubDiscussion.ts

```diff
@@ -0,0 +1,145 @@
+{
+  /**
+   * Create an instance from an IHubDiscussion object
+   * @param json JSON object to create a HubDiscussion from
+   * @param context ArcGIS context
+   * @returns a HubDiscussion
+   */
+  static fromJson(
+    json: Partial<IHubDiscussion>,
+    context: IArcGISContext
+  ): HubDiscussion {
+    // merge what we have with the default values
+    const pojo = this.applyDefaults(json, context);
+    return new HubDiscussion(pojo, context);
+  }
+
+  /**
+   * Create a new HubDiscussion, returning a HubDiscussion instance.
+   * Note: This does not persist the Discussion into the backing store
+   * @param partialDiscussion a partial IHubDiscussion
+   * @param context ArcGIS context
+   * @returns promise that resolves a HubDiscussion
+   */
+  static async create(
+    partialDiscussion: Partial<IHubDiscussion>,
+    context: IArcGISContext,
+    save: boolean = false
+  ): Promise<HubDiscussion> {
+    const pojo = this.applyDefaults(partialDiscussion, context);
+    // return an instance of HubDiscussion
+    const instance = HubDiscussion.fromJson(partialDiscussion, context);
+    if (save) {
+      await instance.save();
+    }
+    return instance;
+  }
+
+  /**
+   * Fetch a Discussion from the backing store and return a HubDiscussion instance.
+   * @param identifier slug or item id
+   * @param context ArcGIS context
+   * @returns promise that resolves a HubDiscussion
+   */
+  static async fetch(
+    identifier: string,
+    context: IArcGISContext
+  ): Promise<HubDiscussion> {
+    try {
+      const entity = await fetchDiscussion(identifier, context.requestOptions);
+      // create an instance of HubDiscussion from the entity
+      return HubDiscussion.fromJson(entity, context);
+    } catch (ex) {
+      throw (ex as Error).message ===
+        "CONT_0001: Item does not exist or is inaccessible."
+        ? new Error("Discussion not found.")
+        : ex;
+    }
+  }
+
+  private static applyDefaults(
+    partialDiscussion: Partial<IHubDiscussion>,
+    context: IArcGISContext
+  ): IHubDiscussion {
+    // ensure we have the orgUrlKey
+    if (!partialDiscussion.orgUrlKey) {
+      partialDiscussion.orgUrlKey = context.portal.urlKey;
+    }
+    // extend the partial over the defaults
+    const pojo = {
+      ...DEFAULT_DISCUSSION,
+      ...partialDiscussion,
+    } as IHubDiscussion;
+    return pojo;
+  }
+
+  /**
+   * Apply a new state to the instance
+   * @param changes A partial IHubDiscussion
+   */
+  update(changes: Partial<IHubDiscussion>): void {
+    if (this.isDestroyed) {
+      throw new Error("HubDiscussion is already destroyed.");
+    }
+    // merge partial onto existing entity
+    this.entity = { ...this.entity, ...changes };
+  }
+
+  async save(): Promise<void> {
+    if (this.isDestroyed) {
+      throw new Error("HubDiscussion is already destroyed.");
+    }
+
+    if (this.entity.id) {
+      // update it
+      this.entity = await updateDiscussion(
```

Comment:
Quoting Tom from [his own PR](https://github.com/Esri/hub.js/pull/993#discussion_r1134363869) here, and wondering if we should be lazy loading these "edit" functions (same with `createDiscussion` and `deleteDiscussion` below):

> We do this so that the edit code is lazily loaded. Most users will only be viewing content (or any other entity). They should not have to wait for edit code to be downloaded and parsed before being able to interact w/ the application.
> 
> Another way to handle this would be to require that the entity classes are only imported from other modules that are themselves lazily loaded (i.e. Stencil components), but that is not currently the case (we import the classes in Ember routes) and would be hard to enforce.
> 
> So for now at least, all entity classes should use dynamic import() for the edit code.